maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/invoker/equipped/InvokerEquippedEmbeddedMavenRunningAsDaemonTestCase.java

```diff
@@ -0,0 +1,65 @@
+package org.jboss.shrinkwrap.resolver.impl.maven.embedded.invoker.equipped;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.PrintStream;
+import java.util.Arrays;
+import org.apache.maven.shared.invoker.DefaultInvocationRequest;
+import org.apache.maven.shared.invoker.DefaultInvoker;
+import org.apache.maven.shared.invoker.InvocationRequest;
+import org.apache.maven.shared.invoker.Invoker;
+import org.assertj.core.api.Assertions;
+import org.jboss.shrinkwrap.resolver.api.maven.embedded.EmbeddedMaven;
+import org.jboss.shrinkwrap.resolver.api.maven.embedded.daemon.DaemonBuild;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.jboss.shrinkwrap.resolver.impl.maven.embedded.Utils.getPropertiesWithSkipTests;
+import static org.jboss.shrinkwrap.resolver.impl.maven.embedded.Utils.pathToJarSamplePom;
+
+public class InvokerEquippedEmbeddedMavenRunningAsDaemonTestCase {
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+
+    @Before
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+    }
+
+    @Test
+    public void testWhenDaemonIsUsedEndOfTheBuildIsNotReached() throws InterruptedException {
+
+        final InvocationRequest request = new DefaultInvocationRequest();
+        Invoker invoker = new DefaultInvoker();
+
+        request.setPomFile(new File(pathToJarSamplePom));
+        request.setGoals(Arrays.asList(new String[] { "clean", "package" }));
+
+        request.setProperties(getPropertiesWithSkipTests());
+
+        DaemonBuild daemonBuild = EmbeddedMaven
+            .withMavenInvokerSet(request, invoker)
+            .useAsDaemon()
+            .build();
+
+        Thread.sleep(900);
```

Comment:
I wasn't sure if it makes sense to add another dependency just because of two lines of code. It was proposed by Alex as well - now, you are two against one, so I should probably change it :-)